packages/scene-composer/src/utils/sceneAssetUtils.ts

```diff
@@ -0,0 +1,40 @@
+import path from 'path';
+
+import { AssetType, ModelFileTypeList } from '../interfaces';
+import { DisplayMessageCategory, IDisplayMessage } from '../store/internalInterfaces';
+
+import { extractFileNameExtFromUrl } from './pathUtils';
+
+export const TILESET_JSON = 'tileset.json';
+
+export const evaluateModelType = (
+  filePath: string,
+  addMessages: (messages: IDisplayMessage[]) => void,
+): { modelName: string; modelType: AssetType } | undefined => {
+  const [name, ext] = extractFileNameExtFromUrl(filePath);
+  const modelType: AssetType = ext.toUpperCase() as AssetType;
+
+  const fileName = `${name}.${ext}`;
+
+  // Tiles3D asset type
+  if (fileName === TILESET_JSON) {
+    // Node name is the parent directory name
+    const dir = path.dirname(filePath);
+    const modelName = dir === '.' ? name : dir.split('/').pop();
+    return {
+      modelName: modelName!,
+      modelType: AssetType.TILES_3D,
+    };
+  }
+
+  if (!ModelFileTypeList.includes(modelType)) {
+    const messageText = `ERROR: Model type ${modelType} not supported`;
```

Comment:
should we localize this string? also, since this modelType may not be a type of 3D model, the message should be something like `File with extension ${modelType} is not a supported model type`

Reply:
Updated